### ISSUE TITLE
Warning for ignored files on linting

### ISSUE THREAD

**Issue by abhisekp** [eslint/eslint#5623]
**What version of ESLint are you using?**
eslint 2.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint 6.0.0-beta.6

**Please show your full configuration:**
> ### .eslintrc.json

```json
{
    "root": true,
    "extends": "airbnb/base",
    "rules": {
        "indent": [
            2,
            2
        ],
        "quotes": [
            0, "double"
        ],
        "linebreak-style": [
            2, "unix"
        ],
        "semi": [
            2, "always"
        ],
        "no-unused-vars": [
            1, {
                "vars": "all",
                "args": "none"
            }
        ],
        "no-var": [1],
        "no-fallthrough": [1]
    },
    "env": {
        "es6": true,
        "node": true
    },
    "ecmaFeatures": {
        "modules": true
    },
    "globals": {
        "logger": true,
        "log4js": true
    },
    "parser": "babel-eslint"
}
```

> ### .eslintignore

```yml
*.min.js
build/
```

**What did you do? Please include the actual source code causing the issue.**
`eslint **/*.js`

**What did you expect to happen?**
Not to show my ignored files WARNING while linting.

**What actually happened? Please include the actual, raw output from ESLint.**
```sh
MyProjects/NodeApp/md-urls  create-lib-WIP ✗                           0m ↑⚑  ⍉
▶ npm run lint

> md-urls@0.0.3 lint /home/abhisekp/MyProjects/NodeApp/md-urls
> eslint **/*.js


/home/abhisekp/MyProjects/NodeApp/md-urls/bin/cli.js
  5:13  warning  'program' is defined but never used  no-unused-vars

/home/abhisekp/MyProjects/NodeApp/md-urls/build/CONTRIBUTING.js
  0:0  warning  File ignored because of a matching ignore pattern. Use --no-ignore to override

/home/abhisekp/MyProjects/NodeApp/md-urls/build/app.js
  0:0  warning  File ignored because of a matching ignore pattern. Use --no-ignore to override

/home/abhisekp/MyProjects/NodeApp/md-urls/build/index.js
  0:0  warning  File ignored because of a matching ignore pattern. Use --no-ignore to override

/home/abhisekp/MyProjects/NodeApp/md-urls/lib/app.js
  1:5  warning  'x' is defined but never used  no-unused-vars

✖ 5 problems (0 errors, 5 warnings)

```

**Comment by platinumazure** [comment]
Workaround (only practical if you have no rules configured as warnings): Use `--quiet` option to suppress warnings and only print errors.

**Comment by abhisekp** [comment]
@platinumazure i've rules configured as warning and i need those warnings.

**Comment by platinumazure** [comment]
I understand- it's a note for others who find this issue :smile:

**Comment by alberto** [comment]
I think it is working as expected, because of the way you are calling eslint. If you use  `eslint .` instead, the warnings should go away.

**Comment by abhisekp** [comment]
@alberto absolutely. Thanks. 
But why should it gimme warning for ignored files?

**Comment by alberto** [comment]
Because there is a discrepancy between the files you are asking to lint in the command line and your ignored files. I'm actually not sure what should take precedence, but this is something we are discussing in [another thread](https://github.com/eslint/eslint/issues/4828), so I'm closing this one.

**Comment by JaKXz** [comment]
@alberto sorry to continue a closed thread, but I don't really follow the similarities in #4828 - 

I'd prefer not to have to use the `--quiet` flag to suppress ALL warnings when I've specifically listed files in my `.eslintignore` that I want eslint to ignore. Is there a good reason for this behaviour that I'm missing? The docs say it's because eslint is unsure about the files that I want to ignore from linting, but, if I've specified it explicitly in an ignore file I don't see that as an issue. It's like if git warned me about ignoring files in my `.gitignore` every time I wanted to commit.

**Comment by alberto** [comment]
Yes, by using `eslint **/*.js` (without quotes) you are explicitly including all files and asking `eslint` to lint them. `eslint` is trying to be helpful telling you that it is not linting some of the files you asked him because they are ignored.

Try using `eslint .` instead.

**Comment by JaKXz** [comment]
Thanks! That makes sense.

**Comment by jhliberty** [comment]
Can we add a rule to not display a warning when an ignored file is linted? It's really frustrating to save an ignored file and see warnings thrown.

![screenshot](https://cloud.githubusercontent.com/assets/449321/15560603/c065d7d6-22b2-11e6-9b09-2bad49c164b2.png)

**Comment by joaoreynolds** [comment]
^ I agree. I don't see any value in having this warning to begin with since the point of gitignore is to ignore files that fall within the group of files you're linting.

**Comment by julbaxter** [comment]
@alberto What's the point of an ignore file if it's not to contradict what you express in the config file?

**Comment by johnomalley** [comment]
Yeah I just complained to JetBrains about this issue... the only workaround to prevent this issue from firing in IDEA/WebStorm is to turn off ESLint integration.  Please reconsider this... it's a really dumb idea to 'warn' the user that the system is working as designed.

**Comment by brandonsturgeon** [comment]
I'd also like to support another consideration for this issue. If I've explicitly ignored a file in `.eslintignore`, I don't need to hear about it anymore.

**Comment by Alexsey** [comment]
@johnomalley could you please link your issue in WebStorm issue tracker?

**Comment by johnomalley** [comment]
https://youtrack.jetbrains.com/issue/WEB-27100

They are claiming that it's fixed... although if that's true it hasn't been released yet.

**Comment by jitendravyas** [comment]
@alberto I'm using grunt plugin https://github.com/sindresorhus/grunt-eslint

Don't know how to use `eslint .`  Can you please help?
